src/solvers/thermal_operators.hpp

```diff
@@ -76,14 +71,14 @@ class DynamicConductionOperator : public mfem::TimeDependentOperator {
   std::unique_ptr<mfem::HypreParMatrix> T_e_mat_;
 
   /**
-   * @brief Assembled RHS vector
+   * @brief Non-owning ptr to assembled RHS vector
    */
-  std::shared_ptr<mfem::Vector> rhs_;
+  mfem::Vector* rhs_ = nullptr;
```

Comment:
Changing this from a shared_ptr (which it is) to a unique_ptr+"non-owning ptr" is just abusing/stretching of the definition of unique.  Can you provide a good explanation of the benefit of this.  To me this just obfuscates the life-cycle of this pointer as well as opens us up to using a pointer that has been freed (even if it doesn't at the moment).

Reply:
I do commend your very explicit non-ownership description.